2-Draft-Accepted/RFCNNNN-PSCore6-and-Windows-Modules.md

```diff
@@ -0,0 +1,81 @@
+---
+RFC: RFCNNNN
+Author: Steve Lee
+Status: Draft
+SupercededBy: n/a
+Version: 0.1
+Area: Engine
+Comments Due: 6/30
+Plan to implement: Yes
+---
+
+# Enable Discovery and Importing of Compatible in-box Windows PowerShell modules
+
+The biggest factor preventing existing Windows PowerShell users from moving to PowerShell Core 6 has been cmdlet coverage.
+Specifically being able to use existing cmdlets that ship with Windows.
+The PowerShell team has been working with Windows teams to port their modules so that they work in PowerShell Core 6.
+This RFC addresses the experience for the user when using PowerShell Core 6.1 with next version of Windows 10.
+
+## Motivation
+
+    As a ITPro,
+    I can use existing Windows PowerShell modules in PowerShell Core 6,
+    so that I can start deploying PowerShell Core 6 in my enterprise.
+
+## Specification
+
+The module manifest (since Windows PowerShell 5) may contain a `CompatiblePSEditions` property to indicate if it is compatible with
+Windows PowerShell (value of `Desktop` even on Windows Server) or PowerShell Core (value of `Core`).
+Currently, this property is only documentation and not observed by PowerShell.
+The proposal is to modify PowerShell Core 6.1 to respect this property.
+
+### Windows PSModulePath
+
+Windows PowerShell, by default, has module paths in:
+
+  1. User profile: $env:UserProfile\Documents\WindowsPowerShell\Modules
+  2. Program files: $env:ProgramFiles\WindowsPowerShell\Modules
+  3. System32: $env:Windir\System32\WindowsPowerShell\v1.0\Modules
+
+The proposed change is to add *only* the `System32` path to `PSModulePath` on PowerShell Core 6.
+
+Modules in the user profile should be re-installed under PowerShell Core 6.
+Duplicate modules is something that can be addressed in PowerShellGet and outside the scope of this RFC.
+Modules in `Program Files` that may be compatible with PowerShell Core 6 is addressed below.
+
+Users who want to search the entirety of the Windows PowerShell PSModulePath should use `Add-WindowsPSModulePath` cmdlet
+from the `WindowsCompatibility` module.
+
+### PowerShell Core PSModulePath
+
+Modules in the PowerShell Core PSModulePaths are implicitly deemed compatible even if the manifest does not
```

Comment:
Does "PowerShell Core PSModulePaths" here refer to:
1. All paths in `$env:PSModulePath` (i.e. configured by the user)?
2. The default PowerShell Core module paths - being the ones that are part of `$env:PSModulePath` in a fresh PowerShell Core installation?
3. (2), but without the `System32` directory proposed in this RFC?

For 2/3, the best documentation I can find for what these paths actually are is in the code in [ModuleIntrinsics.cs](https://github.com/PowerShell/PowerShell/blob/8805ca20aae583b1e63d9cfd2a2a3b24e0862428/src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs#L821-L823), which lists the "PersonalModulePath", "SharedModulePath" and "PSHomeModulePath". Are those the module paths to declare implicitly compatible?

Would it be possible to enumerate the specific paths that are:
* to be module paths by default, and
* to be implicitly compatible module paths

in this RFC (in the same way as the Windows PowerShell module paths above)?

Reply:
`PowerShell Core PSModulePaths` refers to all paths that are part of `$env:PSModulePath` except `System32`.  If the user changed the system `PSModulePath` env var, we should respect that.  There is a separate PR that @BrucePay is authoring for `PSModulePath6` that is outside the scope of this RFC.

The tl;dr; version is that we are special casing the `System32` path in PSCore6.  The existing `PSModulePath` should behave as it behaves today.

Reply:
So any path found in PowerShell Core's `$env:PSModulePath` should be considered implicitly compatible?

Reply:
Yes.